keel-core/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DockerArtifact.kt

```diff
@@ -16,14 +18,19 @@ data class DockerArtifact(
   override val name: String,
   override val deliveryConfigName: String? = null,
   override val reference: String = name,
-  val tagVersionStrategy: TagVersionStrategy = SEMVER_TAG,
+  val tagVersionStrategy: TagVersionStrategy? = null,
   val captureGroupRegex: String? = null,
-  override val from: ArtifactOriginFilterSpec? = null
+  val branch: String? = null,
+  override val from: ArtifactOriginFilterSpec? =
+    branch?.let { ArtifactOriginFilterSpec(BranchFilterSpec(name = branch)) }
 ) : DeliveryArtifact() {
   init {
     require(name.count { it == '/' } <= 1) {
       "Docker image name has more than one slash, which is not Docker convention. Please convert to `organization/image-name` format."
     }
+    require(from == null || tagVersionStrategy == null) {
```

Comment:
This should prevent people from submitting delivery configs with the two formats mixed going forward. There are occurrences in the database that I plan to fix (could add a migration to this PR too).

Reply:
You can use `xor` instead of `||` to enforce exactly one here.

Reply:
I remembered that! I tried it, but I think I didn't get the syntax right. How would that expression look?

Reply:
Done!